db/migrate/20230502092105_create_post.rb

```diff
@@ -0,0 +1,13 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.integer :author_id
```

Comment:
- [ ] Let's make sure that the data in our migration files reflect what we have on the ERD. In this case, we want to name our user foreign keys as __author_id__. By default, this can bring issues because Rails only knows a user (user_id). We need to explicitly tell Rails that a user will be called __author__ and its __id__ will now be __author_id__, __HINT__:
```ruby
    t.references :author, foreign_key: { to_table: 'users' } # This will create author_id and a corresponding index 
```
[Read this documentation for reference](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference)

Reply:
as per project requirement, I have to have a foreign key column `author_id` which references `users(id)`. Which I did in my third migration. Please explain what should have been done. 

![image](https://user-images.githubusercontent.com/116718155/235701667-7c849257-bc3d-49fb-93ef-1eba6dae693a.png)
![image](https://user-images.githubusercontent.com/116718155/235702077-cbdb7365-56d7-4f35-ac7a-3c5a9d4ba05b.png)